pkg/engine/datasource/graphql_datasource/graphql_datasource_test.go

```diff
@@ -250,6 +250,280 @@ func TestGraphQLDataSource(t *testing.T) {
 		},
 	}))
 
+	t.Run("Query with array input", RunTest(subgraphTestSchema, `
+		query($representations: [_Any!]!) {
+			_entities(representations: $representations){
+				... on Product {
+					reviews {
+						body 
+						author {
+							username 
+							id
+						}
+					}
+				}
+			}
+		}
+	`, "", &plan.SynchronousResponsePlan{
+		Response: &resolve.GraphQLResponse{
+			Data: &resolve.Object{
+				Fetch: &resolve.SingleFetch{
+					DataSource: &Source{},
+					BufferId:   0,
+					Input:      `{"method":"POST","url":"https://subgraph-reviews/query","body":{"query":"query($representations: [_Any!]!){_entities(representations: $representations){__typename ... on Product {reviews {body author {username id}}}}}","variables":{"representations":$$0$$}}}`,
```

Comment:
For a moment I thought that existing entity queries might somehow be broken, but that didn't seem right. This test appears to be testing a scenario where the engine itself receives an entity query. I don't think this is realistic since the [federation spec](https://www.apollographql.com/docs/federation/federation-spec/#query_service) indicates that federation modifications SHOULD NOT be included in service SDLs. That means _entites and the related types would never appear in a plan data source configuration.

I think this also means the _Any scalar wouldn't necessarily the the federation _Any type (though I suppose if someone defined a type with the same name but with different semantics that would conflict with federation and be really confusing... and why would someone define a custom underscore type anyway?).

Reply:
We actually have a use case for this. If all subgraphs are a managed API themselves, we use the engine as a "proxy" to route traffic to the subgraph. In this scenario, we have one + N engines. One represents the supergraph, N the subgraphs. This way, you can manage supergraphs and subgraphs independently.

Reply:
Adding "_Any" is a good shortcut until we find a better solution I guess, e.g. allowing a datasource to add their own scalars or schema extensions, including validation hooks..